### DISCUSSION TITLE
New Oauth2 API (using popup) problem using Google.

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2375] [Q&A] [answered]
I'm having trouble with the new oauth2 signin API using Google.  The old way was working. I added the new "Authorized redirect URIs": `https://DOMAIN/api/oauth2-redirect`

I don't see a domain in the URI of the error message.

```
Error 400: invalid_request
Request details: redirect_uri=/api/oauth2-redirect 
```

**Comment by ganigeorgiev** [comment, marked as answer]
Do you use relative url as part of the SDK initialization? If yes please try specifying the absolute url, aka.:

```js
const pb = new PocketBase("https://YOUR_DOMAIN");
```

_I'll consider sometime later today to add an option to build the absolute url (at least in the browser) when constructing the redirect url._

**Reply by dschissler** [reply]
Yes that was it.  Thanks.

**Reply by dschissler** [reply]
cool

https://github.com/pocketbase/js-sdk/releases/tag/v0.14.1

**Reply by dschissler** [reply]
@ganigeorgiev Is it not possible to get a collection `expand` using the new `authWithOAuth2` method?  I noticed that inside of it `authWithOAuth2Code` only is called with 5 arguments, but the 7th argument is where the `expand` would go.

**Reply by ganigeorgiev** [reply]
We can add `query` prop to the `OAuth2AuthConfig` interface and pass it down to the internal `authWithOAuth2Code`. I'll consider updating it later today/tomorrow.

**Reply by dschissler** [reply]
Cool.  I used it for group membership to give a friendly UI message or redirect if the user went there but somehow doesn't have permission.  A hard refresh would redirect from the server.

**Reply by ganigeorgiev** [reply]
I've pushed a JS SDK update as v0.14.3 release (_the Dart SDK already has the `expand`, and the new `fields` params that will be available with PocketBase v0.16.0, as optional named args_).

```js
await pb.collection('users').authWithOAuth2({
    provider: 'github',
    query: {
        expand: 'someRelField',
    },
});
```
